### PR TITLE
Zero Shot Classification with Hugging Face

### PR DESCRIPTION
**Pull request by vitalii-bulyzhyn**

### CODE REVIEW

File: eventhandlers/action_runners/actions/HuggingFaceZeroShot.py [hunk 1]
```diff
@@ -2,15 +2,20 @@
...
+from shared.database.annotation.instance import Instance
+from shared.database.project import Project
+from shared.database.task.task import Task
 from shared.shared_logger import get_shared_logger
-from shared.database.task.job.job import Job
-from shared.helpers.sessionMaker import session_scope
-from shared.utils import job_dir_sync_utils
+from shared.database.task.task import Task
 from shared.database.source_control.file import File
+from shared.database.annotation.instance import Instance
```
Comment (anthony-sarkis) [inline eventhandlers/action_runners/actions/HuggingFaceZeroShot.py:11]: duplicate imports

File: eventhandlers/action_runners/actions/HuggingFaceZeroShot.py [hunk 2]
```diff
@@ -34,78 +39,87 @@ def test_execute_action(self, session, file_id, connection_id):
...
-            logger.warning(f'Action has no file_id Stopping execution')
-            return
-    
+        project_id = self.action.config_data['project_id']
+        group_id = self.action.config_data['group_id']
+
+        task = None
+        task_id = None
+
+        if event_name == 'task_created':
```
Comment (anthony-sarkis) [inline eventhandlers/action_runners/actions/HuggingFaceZeroShot.py:50]: refactor to generic for actions?
is task id not trusted here?

File: eventhandlers/action_runners/actions/HuggingFaceZeroShot.py [hunk 3]
```diff
@@ -34,78 +39,87 @@ def test_execute_action(self, session, file_id, connection_id):
...
-                """
-
+        event_name = self.action.trigger_data.get('event_name')
         file_id = self.event_data['file_id']
-        if not file_id:
-            logger.warning(f'Action has no file_id Stopping execution')
-            return
-    
+        project_id = self.action.config_data['project_id']
+        group_id = self.action.config_data['group_id']
```
Comment (anthony-sarkis) [inline eventhandlers/action_runners/actions/HuggingFaceZeroShot.py:45]: group could mean anything, clarify attribute group
I'd like to learn more on plan for handling this type of "sub schema" selection - maybe something more general to talk about

File: eventhandlers/action_runners/actions/HuggingFaceZeroShot.py [hunk 4]
```diff
@@ -34,78 +39,87 @@ def test_execute_action(self, session, file_id, connection_id):
...
+
+        candidate_attributes = [option['name'] for option in group_list_serialized[0]['attribute_template_list']]
         classifier = pipeline("zero-shot-classification")
 
-        result = classifier(raw_text, candidate_labels)
+        result = classifier(text, candidate_attributes)
 
-        print(result)
+        attribute_to_apply = result['labels'][result['scores'].index(max(result['scores']))]
+        attribute_item_to_apply = [option for option in group_list_serialized[0]['attribute_template_list'] if option['name'] == attribute_to_apply][0]
```
Comment (anthony-sarkis) [inline eventhandlers/action_runners/actions/HuggingFaceZeroShot.py:91]: lets unpack this line it's doing too much

**Review by PJEstrada — APPROVED** [review]

File: eventhandlers/action_runners/actions/HuggingFaceZeroShot.py (hunk 3 above)
Comment (vitalii-bulyzhyn) [reply eventhandlers/action_runners/actions/HuggingFaceZeroShot.py:45]: I agree that group_id isn't quite descriptive, the reason is it named that way is I was keeping the naming as it is on the other parts of the diffgram

File: eventhandlers/action_runners/actions/HuggingFaceZeroShot.py (hunk 2 above)
Comment (vitalii-bulyzhyn) [reply eventhandlers/action_runners/actions/HuggingFaceZeroShot.py:47]: What do you mean by "not trusted"? If you mean that it's optional - then yes, if the trigger is input_file_uploaded - then action will be executed without task_id

File: eventhandlers/action_runners/actions/HuggingFaceZeroShot.py (hunk 3 above)
Comment (anthony-sarkis) [reply eventhandlers/action_runners/actions/HuggingFaceZeroShot.py:42]: well then we should change those other parts too eventually
it should only be group if it's super clear it's in the attribute context

**Pull request merged**